examples/nidaqmx_analog_input/measurement.py

```diff
@@ -15,34 +25,67 @@
+@measurement_service.configuration(
+    "pin_name",
+    nims.DataType.Pin,
+    "Pin1",
+    instrument_type=nims.session_management.INSTRUMENT_TYPE_NI_DAQMX,
+)
 @measurement_service.configuration("sample_rate", nims.DataType.Double, 1000.0)
 @measurement_service.configuration("number_of_samples", nims.DataType.UInt64, 100)
 @measurement_service.output("acquired_samples", nims.DataType.DoubleArray1D)
-def measure(physical_channel, sample_rate, number_of_samples):
+def measure(pin_name, sample_rate, number_of_samples):
     """Perform a finite analog input measurement with NI-DAQmx."""
     logging.info(
-        "Executing measurement: physical_channel=%s sample_rate=%g number_of_samples=%d",
-        physical_channel,
+        "Executing measurement: pin_name=%s sample_rate=%g number_of_samples=%d",
+        pin_name,
         sample_rate,
         number_of_samples,
     )
+    session_management_client = nims.session_management.Client(
+        grpc_channel=measurement_service.get_channel(
+            provided_interface=nims.session_management.GRPC_SERVICE_INTERFACE_NAME,
+            service_class=nims.session_management.GRPC_SERVICE_CLASS,
+        )
+    )
+    with contextlib.ExitStack() as stack:
+        reservation = stack.enter_context(
+            session_management_client.reserve_sessions(
+                context=measurement_service.context.pin_map_context,
+                pin_or_relay_names=[pin_name],
+                instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DAQMX,
+                # If another measurement is using the session, wait for it to complete.
+                # Specify a timeout to aid in debugging missed unreserve calls.
+                # Long measurements may require a longer timeout.
+                timeout=60,
+            )
+        )
+        if len(reservation.session_info) != 1:
+            measurement_service.context.abort(
+                grpc.StatusCode.INVALID_ARGUMENT,
+                f"Unsupported number of sessions: {len(reservation.session_info)}",
+            )
+        session_info = reservation.session_info[0]
+
+        def cancel_callback(task: Optional[nidaqmx.Task] = None):
+            logging.info("Canceling measurement")
+            task_to_abort = task
+            if task_to_abort is not None:
+                task_to_abort.control(TaskMode.TASK_ABORT)
 
-    def cancel_callback():
-        logging.info("Canceling measurement")
-        task_to_abort = task
-        if task_to_abort is not None:
-            task_to_abort.control(nidaqmx.constants.TaskMode.TASK_ABORT)
+        measurement_service.context.add_cancel_callback(cancel_callback)
 
-    measurement_service.context.add_cancel_callback(cancel_callback)
-    time_remaining = measurement_service.context.time_remaining
+        time_remaining = measurement_service.context.time_remaining
+        timeout = min(time_remaining, 10.0)
```

Comment:
In https://github.com/ni/measurementlink-python/pull/268#discussion_r1188651878 I listed this as step 4 to indicate that it doesn't belong above step 3 (create task).

The best place for this code is with task.read(), and it doesn't need to be split across two lines.
```
timeout = min(measurement_service.context.time_remaining, 10.0)
voltage_values = task.read(number_of_samples_per_channel=number_of_samples, timeout=timeout)
```

This is better because it accounts for the time taken by create_task, add_ai_voltage_chan, and cfg_samp_clk_timing. Also, defining the value near where it's used often makes code more readable.

Reply:
Ok. Updated.